crates/telio-model/src/features.rs

```diff
@@ -596,857 +616,374 @@ impl Features {
-            {
-                "priority": ["relay", "direct"],
-                "force": "relay"
-            },
-            "direct": {},
-            "dns": {
-                "exit_dns": {}
-            }
-        }"#;
+    mod deserialization {
+        use super::*;
+
+        #[test]
+        fn test_empty_json() {
+            let json = "{}";
+            let features = from_str::<Features>(json).unwrap();
+            assert_eq!(features, Features::default());
+        }
 
-    const CORRECT_FEATURES_JSON_WITH_ALL_OPTIONAL: &str = r#"
+        #[test]
+        fn test_all_fields_with_non_default_values() {
+            let json = r#"
         {
-            "wireguard":
-            {
+            "wireguard": {
                 "persistent_keepalive": {
-                    "vpn": null,
-                    "stun": 50
+                    "vpn": 1,
+                    "direct": 2,
+                    "proxying": 3,
+                    "stun": 4
                 }
             },
-            "nurse": {},
-            "lana":
-            {
-                "event_path": "path/to/some/event/data",
+            "nurse": {
+                "fingerprint": "test_fingerprint",
+                "heartbeat_interval": 5,
+                "initial_heartbeat_interval": 6,
+                "qos": {
+                    "rtt_interval": 7,
+                    "rtt_tries": 8,
+                    "rtt_types": ["Ping"],
+                    "buckets": 9
+                },
+                "enable_nat_type_collection": true,
+                "enable_relay_conn_data": false,
+                "enable_nat_traversal_conn_data": false,
+                "state_duration_cap": 10
+            },
+            "lana": {
+                "event_path": "some/test/path.db",
                 "prod": true
             },
-            "paths":
-            {
-                "priority": ["relay", "direct"],
+            "paths": {
+                "priority": ["direct"],
                 "force": "relay"
             },
-            "direct":
-            {
-                "providers": 42,
-                "endpoint_interval_secs": 10,
-                "skip_unresponsive_peers":
-                {
-                    "enabled": true,
-                    "no_rx_threshold_secs": 50
+            "direct": {
+                "providers": ["local"],
+                "endpoint_interval_secs": 11,
+                "skip_unresponsive_peers": {
+                    "no_rx_threshold_secs": 12
+                },
+                "endpoint_providers_optimization": {
+                    "optimize_direct_upgrade_stun": false,
+                    "optimize_direct_upgrade_upnp": false
                 }
             },
             "is_test_env": true,
-            "derp":
-            {
-                "tcp_keepalive": 1,
-                "derp_keepalive": 2,
-                "enable_polling": true
+            "derp": {
+                "tcp_keepalive": 13,
```

Comment:
Just wondering. Why did you change the values here ?

Reply:
I created a new test and decided to make the numbers always increase by one, and then removed the old test without really looking at what values were there